src/main/scala/tech/cryptonomic/conseil/tezos/TezosDatabaseOperations.scala

```diff
@@ -755,19 +757,68 @@ object TezosDatabaseOperations extends LazyLogging {
   def writeProcessedEventsLevels(eventType: String, levels: List[BigDecimal]): DBIO[Option[Int]] =
     Tables.ProcessedChainEvents ++= levels.map(Tables.ProcessedChainEventsRow(_, eventType))
 
-  /** Inserts registered tokens to the table if empty
-    *
+  /** Inserts to the table if table is empty
+    * @param table slick TableQuery[_] to which we want to insert
     * @param rows the type of event to record
     * @return the number of entries saved added to the
     */
-  def writeRegisteredTokensRowsIfEmpty(
-      rows: List[Tables.RegisteredTokensRow]
+  def insertIfEmptyTable[A <: AbstractTable[_]](
+      table: TableQuery[A],
+      rows: List[A#TableElementType]
   )(implicit ec: ExecutionContext): DBIO[Option[Int]] =
-    Tables.RegisteredTokens.exists.result.flatMap {
+    table.exists.result.flatMap {
       case true => DBIO.successful(Some(0))
-      case false => Tables.RegisteredTokens ++= rows
+      case false => table ++= rows
     }
 
+  import shapeless._
+  import shapeless.ops.hlist._
```

Comment:
😱  😄

Reply:
don't get overexcited, I just used the example from stackoverflow 😛 
https://stackoverflow.com/questions/45885976/scala-is-there-a-generic-way-to-trim-all-string-fields-of-case-classes